addon/components/tui-editor.js

```diff
@@ -1,157 +1,108 @@
-        if (this.get('onLoad')) {
-            this.onLoad();
-        }
-    },
-
-    onChangeWrapper() {
-        if (this.get('onChange')) {
-            this.onChange(this.editor.getValue());
-        }
-    },
-
-    onStateChangeWrapper() {
-        if (this.get('onStateChange')) {
-            this.onStateChange();
-        }
-    },
-
-    onFocusWrapper() {
-        if (this.get('onFocus')) {
-            this.onFocus();
-        }
-    },
-
-    onBlurWrapper() {
-        if (this.get('onBlur')) {
-            this.onBlur();
-        }
+  layout,
+
+  value: '',
+
+  // here we use a syntax like <property>:<method>:<tui option> to update such property (optional)>
+  tuiOptions: Object.freeze([
+    'previewStyle:changePreviewStyle', 'editType:changeMode:initialEditType', 'height:height', 'minHeight:minHeight',
+    'language', 'useDefaultHTMLSanitizer', 'useCommandShortcut', 'codeBlockLanguages', 'usageStatistics',
+    'toolbarItems', 'hideModeSwitch', 'viewer', 'value:setValue:initialValue'
+  ]),
+
+  // gathers all the options to initialize TUI editor, respecting tuiOptions syntax
+  options: computed('tuiOptions.[]', function() {
+    let options = {};
+
+    this.get('tuiOptions').forEach((o) => {
+      let [optionName, ,tuiOption] = o.split(':');
+      tuiOption = tuiOption ? tuiOption : optionName;
+
+      options[tuiOption] = this.get(optionName);
+    });
+
+    return options;
+  }),
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.setupEditor();
+    this.addObservers();
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    this.removeObservers();
+  },
+
+  setupEditor() {
+    let options = this.get('options');
+
+    let editor = Editor.factory(assign(options, {
+      el: this.element,
+      events: {
+        load: (...args) => this.eventInvoked('onLoad', ...args),
+        change: (...args) => this.eventInvoked('onChange', this.editor.getValue(), ...args),
+        stateChange: (...args) => this.eventInvoked('onStateChange', ...args),
+        focus: (...args) => this.eventInvoked('onFocus', ...args),
+        blur: (...args) => this.eventInvoked('onBlur', ...args)
+      }
+    }));
+
+    this.set('editor', editor);
+  },
+
+  // tests if an `actionName` function exists and calls it with the arguments if so
+  eventInvoked(actionName, ...args) {
+    if (this.get(actionName)) {
+      this.get(actionName)(...args);
+    }
+  },
+
+  addObservers() {
+    this._observers = {};
+    this.get('tuiOptions').forEach((o) => {
+      let [optionName, tuiMethod] = o.split(':');
+
+      if (tuiMethod) {
+        this._observers[optionName] = function() {
+          let value = this.get(optionName);
+
+          // `value` is a special case because using `setValue`
+          // moves the current cursor position so we need to avoid calling it
+          // only call it when the editor value is different from the new value we got
+          if (optionName === 'value') {
```

Comment:
```suggestion
          if (optionName === 'value' && this.editor.getValue) {
```
getValue is only available on editor instances, but not on the markdown viewer.